src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let current = { ...state };
```

Comment:
The cloning of the 'state' object is unnecessary in this line as it is already cloned inside the 'switch' statement.